addons/web_editor/static/src/scss/wysiwyg_snippets.scss

```diff
@@ -610,8 +610,10 @@ body.editor_enable.editor_has_snippets {
                     }
                 }
             }
-            we-input, we-input div, we-input input {
-                width: 100% !important;
+            we-input:not(.o_we_small_input) {
+                & div, & input {
+                    width: 100% !important;
+                }
```

Comment:
there could be a better solution but i haven't found a better one .. maybe "o_we_large_input" could be used in the toolbar instead of this rule ...
cc @Zinston

Reply:
I'm missing context here. Is this a fix? For what exactly?
For the record, `o_we_large_input` already exists :
```scss
we-input.o_we_user_value_widget.o_we_large_input > div {
    flex: 1 1 auto;
}
```
I don't really know what this `o_we_small_input` is for.

Also watch out because here you actually change the behavior for `we-input:not(.o_we_small_input)` since the original applied to it but with your change it's only for its `div` and `input` _descendants_.

Reply:
Its an imp, I added [some options in the toolbar](https://i.imgur.com/Gz3fd2H.jpg) to animate a selection of text but I don't need these inputs to be full width.
And the actual rule is "input in the toolbar = width: 100% important".

Reply:
Then it seems like a good solution to me.
Are you aware though that [a PR](https://github.com/odoo/odoo/pull/73565) is in merging process, that completely redesigns the toolbar? I checked and luckily that rule is still there so you should be ok but it'll be worth testing again once merged.

Anyway, to address what I previously said about the descendants I'd do something like this, or the rule won't apply to `we-input` itself anymore:

```scss
we-input:not(.o_we_small_input) {
    width: 100% !important;

    & div, & input {
        width: 100% !important;
}
```

Reply:
yes I know for the redesign ;)
I rather did like this, but thank you I missed this error
`we-input:not(.o_we_small_input) {
    &, & div, & input {
        width: 100% !important;
}`

Reply:
Nice, I learned something in the process, didn't know you could do that :-)

Reply:
Should remove the `&` before `div` and `input` though

Reply:
They're unnecessary but not wrong.

Reply:
> They're unnecessary but not wrong though.

Like `true` here: `if (true && condition) { ` :wink:

Reply:
I'm playing with semantics here but I don't think it's the same since iirc `a { b { ... } }` here is shorthand for `a { & b { ... } }`. But yeah sure in any case you're right :-P